chore: adds fathom mcp

viveksingh101994 opened the pull request:
```curl
curl --request POST \
  --url http://0.0.0.0:3000/api/mcp-servers \
  --header 'Content-Type: application/json' \
  --data '
{
  "name": "fathom",
  "type": "stdio",
  "command": "docker",
  "args": [
    "run",
    "-i",
    "--rm",
    "-e",
    "FATHOM_API_KEY",
    "fathom-mcp"
  ],
  "env": {
    "FATHOM_API_KEY": "<env_var"
  },
  "isDisabled": false
}
'
```

NOTES:
- I used https://github.com/Dot-Fun/fathom-mcp and ran it via docker
- I also added mcp server which i created from scratch on node (This one we can run using `node` command in stdio method)


Both of them produce the same result with the code i have written, both of them have pros and cons,

### PROS/CONS
#### https://github.com/Dot-Fun/fathom-mcp
- Pros
   - No need to maintain, just need to update our docker image with the code 
   - We can easily plug-in with uv as well
- Cons
   - We need to modify it if we need to make some changes example running it via http streamable


#### Self hosted code
- Pros
   - Easy to update the code and we can run via `node`
- Cons
   - Maintainance comes on us and we have to maintain the code

viveksingh101994 committed:
chore: adds fathom mcp

buildgreatthings commented:
Yikes, if we have to run yet another MCP server. Why can't we run via stdio locally?

viveksingh101994 committed:
refactor(fathom-mcp-server): migrate to monorepo structure

Restructure fathom-mcp-server by moving it into the packages/ directory
to adopt a monorepo layout. Simplify npm scripts by consolidating dev
and start commands to directly execute TypeScript using tsx, eliminating
the need for a separate dev script and build step during local development.

Changes:
- Move all fathom-mcp-server files to packages/fathom-mcp-server/
- Remove separate 'dev' script from package.json
- Update 'start' script to use tsx for direct TypeScript execution
- Update pnpm-lock.yaml to reflect new package structure

buildgreatthings committed:
fixes for fathom

buildgreatthings committed:
Merge pull request #32 from reality-platforms/mcp-fathom-fixes

fixes for fathom

viveksingh101994 committed:
Merge branch 'github_mcp' of github.com:reality-platforms/ebee-oss into fathom-mcp

TranHung92 commented on packages/shared-util/types/fathom/index.ts:
nitpick: can make use of pnpm workspace to import these types from the fathom-mcp-server package so you don't have to define these twice

viveksingh101994 replied:
https://github.com/reality-platforms/ebee-oss/pull/31/commits/37f12b2881a8bdc6b7c1a97d87d36a73511cd7b0

viveksingh101994 committed:
chore: adds shared-util package
